automation/Extension/firefox/src/privileged/stackDump/OpenWPMStackDumpChild.jsm

```diff
@@ -0,0 +1,175 @@
+"use strict";
+
+var EXPORTED_SYMBOLS = ["OpenWPMStackDumpChild"];
+const { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
+
+class Controller {
+  constructor(actor) {
+    this.actor = actor;
+    Services.obs.addObserver(this, "http-on-opening-request");
```

Comment:
Why do we need to register two observers here?

Reply:
From my understanding it's just copied from [here](https://searchfox.org/mozilla-central/source/devtools/server/actors/network-monitor/stack-trace-collector.js#37) also the [comments in the IDL](https://searchfox.org/mozilla-central/source/netwerk/protocol/http/nsIHttpProtocolHandler.idl#127) indicate that document and http channels are two different things.
Maybe @nhnt11 can expand

Reply:
mmmm, ok, technically I think we can use just the http notification here. document channel is a more abstract channel type that exists before we're sure that this is an http channel. I'm not entirely sure why the devtools stack trace collector is using this. we should verify that removing it doesn't do anything bad first though, for sanity.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1556489#c12
As per this comment this was intentional, so I'd just keep it as is